site/docs/toolchains.md

```diff
@@ -25,27 +26,34 @@ toolchain most appropriate to that build. It does so by matching the
 [constraints](platforms.html#defining-a-platform) specified in the project's
 `BUILD` file(s) with the constraints specified in the toolchain definition.
 
-**Note:** Some Bazel rules do not yet support toolchain resolution.
-
-When a target requests a toolchain, Bazel checks the list of registered
-toolchains and creates a dependency from the target to the first matching
-toolchain it finds. To find a matching toolchain, Bazel does the following:
-
-1.  Looks through the registered toolchains, first from the `--extra_toolchains`
-    flag, then from the `register_toolchains` calls in the project's `WORKSPACE`
-    file.
-
-2.  For each registered toolchain, Bazel performs the following checks:
-
-    a. Does the toolchain match the requested toolchain type? If not, skip it.
-
-    b. Do the toolchain's execution and target constraints match the constraints
-       stated in the project's execution and target platforms? If yes, the
-       toolchain is a match.
+## Toolchain Resolution
+
+Toolchain resolution is the process of examining the target platform, required
+toolchain types, and available execution platforms, to choose the correct
+execution platform and toolchains for that configured target. Toolchain
+resolution proceeds by the following algorithm:
+
+Inputs: The types of toolchain required (there can be zero, one, or more), the target platform
```

Comment:
This is out of place and awkwardly worded. I'd replace with "Inputs to the resolution mechanism include the required toolchain types (including none) and the target platform." Place it before the last sentence "Toolchain resolutions works as follows:"

Reply:
Done.